src/adapters/node.js

```diff
@@ -8,11 +7,20 @@
         defer: Q.defer,
         http: function(args) {
             var deff = Q.defer();
-            args.body = args.data;
+            if(args.data && typeof args.data === "string") {
+              // Setting args.json requires args.body to be a JSON as per request docs.
+              args.body = JSON.parse(args.data);
```

Comment:
Need to use `try {} catch (e) {}` construction for JSON.parse, string may is not JSON format

Reply:
You can make custom error if string is wrong JSON

Reply:
If we catch the error we should through it back to caller. I am trying to understand the suggestion. Are you saying we should give the caller a better message and/or log the error?

Reply:
Yes, we can send better message for caller.
Like this:
```
try {
  args.body = JSON.parse(args.data);
} catch (e) {
  throw new Error('...string is not JSON', e);
}
```

Reply:
That will mask the original stacktrace...

Reply:
why?

if we don't catch error:
```
SyntaxError: Unexpected token a in JSON at position 0
    at JSON.parse (<anonymous>)
    at http (/Users/mike1pol/work/hs/fhir.js/src/adapters/node.js:12:20)
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/http.js:9:67
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/search.js:182:20
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:25:24
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:68:24
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
```
if we catch error
```
Error: data is not a JSON
    at http (/Users/mike1pol/work/hs/fhir.js/src/adapters/node.js:15:11)
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/http.js:9:67
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/search.js:182:20
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:25:24
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:68:24
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/core.js:58:28
    at /Users/mike1pol/work/hs/fhir.js/src/middlewares/http.js:27:20

```

Reply:
The line 12 is the original error. If there are more lines added to try block, then it is a guess work to pin point the error.

In any case it is not my main concern. I can go with the change. Please let me know, if you have other suggestions.